tests/vnc_test.go

```diff
@@ -166,6 +128,34 @@ var _ = Describe("[rfe_id:127][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 			_, err = wrappedRoundTripper.RoundTrip(req)
 			Expect(err).ToNot(HaveOccurred())
 		})
+
+		DescribeTable("Dual connection. Validate default behavior and with 'preserve-session' option", func(preserve1, preserve2 bool) {
+			ctx1, cancel1, connected1 := newCtxWithConnect()
+			DeferCleanup(cancel1)
+			go func() {
+				defer GinkgoRecover()
+				vncConnect(ctx1, vmi, preserve1, "")
+			}()
+			Eventually(connected1, vncConnectTimeout).Should(Receive(BeTrue()))
+
+			ctx2, cancel2, connected2 := newCtxWithConnect()
+			DeferCleanup(cancel2)
+			go func() {
+				defer GinkgoRecover()
+				errMsg := ""
+				if preserve2 {
+					errMsg = "websocket: bad handshake"
+				}
+				vncConnect(ctx2, vmi, preserve2, errMsg)
```

Comment:
Would it make sense to put handling of `errMsg` into `vncConnect` and to provide a bool like `successExpected` instead?

Reply:
Hm, at the moment we have only this generic error but I'm hoping to be able to improve that. `websocket: bad handshake` is not helpful at all to the client. I'm hoping we can improve the error message so we would be able to match the actual reason for failure "e.g: `client already connected` or `vmi does not have graphics` or `vmi state is $state, can't connect`

But if you prefer moving this to a boolean now and refactor later (if needed) and don't mind